### PR TITLE
Add MySQL CDC example

### PR DESCRIPTION
**Pull request by nicholastmosher**
This PR adds our MySQL CDC example to our `examples/` directory as a set of standalone crates.

This is also where all of the work for porting the CDC example from the old fluvio client to the new one is taking place.

Some things that are left to do:

- [ ] Make the producer read the latest offset in a partition (blocked on #312)
- [x] In the meantime, I'll make the producer track its own offset in a file on disk
- [x] Simplify and slim down the code as much as possible
- [x] Streamline the setup process as much as possible

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [workspace]
 members = [
     "examples/00-echo",
+    "examples/01-cdc/producer",
+    "examples/01-cdc/consumer",
+    "examples/01-cdc/messages",
```
Comment (sehz) [inline Cargo.toml:6]: shouldn't cdc be in a single crate?

File: examples/01-cdc/README.MD [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+The laptop demo environment with 2+ mysql instances requires Docker:
+
+* [Setup MYSQL Docker Environment](./docker/README.MD)
+
+Please complete the Docker setup, before moving to the next section.
+
+
+## Build Producer/Consumer
+
+In **flv-mysql-cdc-demo** directory, build producer/consumer
```
Comment (sehz) [inline examples/01-cdc/README.MD:30]: is this correct?

File: examples/01-cdc/README.MD [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+```
+$ cargo build
+...
+Compiling cdc-producer v0.1.0 (/projects/github/flv-mysql-cdc-demo/mysql-cdc/cdc-producer)
```
Comment (sehz) [inline examples/01-cdc/README.MD:35]: use example directory?

File: examples/01-cdc/README.MD [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+## Start CDC Producer
+
+**None**: Producer must be started before Consumer, as the producer creates the topic in Fluvio.
+
+Producer **profile** is configured to connect to the mysql database configured in [Docker setup](./docker/README.MD).
+
+Start producer:
+
+```
+./target/debug/cdc-producer ./cdc-producer/producer_profile.toml
```
Comment (sehz) [inline examples/01-cdc/README.MD:49]: use `cargo run`

File: examples/01-cdc/README.MD [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+```
+
+## Start CDC Consumer
+
+Consumer **profile** is configured to connect to the mysql database configured in [Docker setup](./docker/README.MD). 
+
+Start consumer:
+
+```
+./target/debug/cdc-consumer cdc-consumer/consumer_profile.toml
```
Comment (sehz) [inline examples/01-cdc/README.MD:59]: use `cargo run`

File: examples/01-cdc/README.MD [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+
+## Fluvio Events
+
+CDC producer generates the events on the **Fluvio topic** as defined in the producer profile. By default the topic name is **rust-mysql-cdc**. 
+
+To view the events generated by the CDC producer, start run **fluvio consumer** command:
+
+```
+$ ./target/debug/fluvio consume rust-mysql-cdc -B
```
Comment (sehz) [inline examples/01-cdc/README.MD:158]: `cargo run`

File: examples/01-cdc/consumer/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+[package]
```
Comment (sehz) [inline examples/01-cdc/consumer/Cargo.toml:1]: consolidate in a single crate?

File: examples/01-cdc/consumer/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+mysql = "18.2.0"
+crossbeam-channel = "0.4.2"
+toml = "0.5.6"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0.53"
+dirs = "2.0.2"
+ctrlc = "3.1.4"
+bigdecimal = "0.1.2"
+http = "0.2.1"
+flv-future-aio = "2.2.1"
```
Comment (sehz) [inline examples/01-cdc/consumer/Cargo.toml:17]: use `fluvio-future`,  `flv-future-aio` is depecrated

**Review by sehz — CHANGES_REQUESTED** [review]

**Comment by nicholastmosher** [comment]
Sorry @sehz I didn't mean for this to be reviewed yet. I just copied directly from the old demo repo into here and I'm working on cleaning it up incrementally

**Comment by sehz** [comment]
Ok. Wont review until it is ready

**Comment by sehz** [comment]
is this ready?

**Comment by nicholastmosher** [comment]
@sehz not yet, I want to at a minimum consolidate all of the documentation and fix it. I'm sure I messed up some filepaths with all of the moving around. But I want to wait on that until we also have a good Getting Started guide to reference back to so I can make sure it all reads well together.

**Pull request closed without merging**